src/client.cpp

```diff
@@ -733,7 +733,50 @@ namespace quicr {
                         attrs.start_location.object = msg.group_0->standalone.start.object;
                         attrs.end_group = msg.group_0->standalone.end.group;
                         attrs.end_object = msg.group_0->standalone.end.object;
-                        FetchReceived(conn_ctx.connection_handle, msg.request_id, tfn, attrs);
+                        if (!FetchReceived(conn_ctx.connection_handle, msg.request_id, tfn, attrs)) {
+                            // TODO: We don't really know enough from this return value to give a specific error.
+                            SendFetchError(conn_ctx, msg.request_id, messages::FetchErrorCode::kInternalError, "");
+                        }
+                        return true;
+                    }
+                    case messages::FetchType::kAbsoluteJoiningFetch: {
+                        [[fallthrough]];
+                    }
+                    case messages::FetchType::kRelativeJoiningFetch: {
+                        const auto& joining = msg.group_1->joining;
+                        const auto sub = conn_ctx.sub_tracks_by_request_id.find(joining.request_id);
+                        if (sub == conn_ctx.sub_tracks_by_request_id.end()) {
+                            SendFetchError(conn_ctx,
+                                           msg.request_id,
+                                           messages::FetchErrorCode::kInvalidJoiningRequestId,
+                                           "Joining track not found");
+                            return true;
+                        }
+                        const auto location = sub->second->GetLatestLocation();
+                        if (!location.has_value()) {
+                            SendFetchError(conn_ctx,
+                                           msg.request_id,
+                                           messages::FetchErrorCode::kInvalidRange,
+                                           "No objects published");
+                            return true;
+                        }
+                        messages::GroupId start_group;
+                        if (msg.fetch_type == messages::FetchType::kAbsoluteJoiningFetch) {
+                            start_group = joining.joining_start;
```

Comment:
Draft 14  says: 

> For an Absolute Joining Fetch, the publisher sets the Start Location to Joining Start.

This would include object as well.  For relative, it does knind of indicate that object is zero based on `{Subscribe Largest Location.Group - Joining Start, 0}`